lib/smart_properties.rb

```diff
@@ -108,13 +108,15 @@ def included(base)
   # @param [Hash] attrs the set of attributes that is used for initialization
   #
   def initialize(*args, &block)
+    properties = self.class.properties.to_hash
```

Comment:
For performance reasons ;) `PropertyCollection` doesn't implement caching, because then every `PropertyCollection` would have to track it's children and notify them if a class was extended with a property at runtime. It's a rare use case, but Ruby has a open class sytem so I want to support that feature. `PropertyCollection#to_hash` results in walking the `PropertyCollection` chain and merging potentially multiple hashes.